File: chazm-model/src/main/java/runtimemodels/chazm/model/notification/DefaultPublisher.java
package runtimemodels.chazm.model.notification;

import javax.inject.Inject;
import javax.inject.Singleton;
import javax.validation.constraints.NotNull;

@Singleton
class DefaultPublisher implements Publisher {

    private final Mediator mediator;

    @Inject
    DefaultPublisher(@NotNull final Mediator mediator) {
        this.mediator = mediator;
    }

    @Override
    public <T> void post(@NotNull final T event) {
        mediator.post(event);
    }

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/notification/DefaultPublisher.kt
package runtimemodels.chazm.model.notification

import javax.inject.Inject
import javax.inject.Singleton

@Singleton // TODO to be replaced with RxJava
internal open class DefaultPublisher @Inject constructor(private val mediator: Mediator) : Publisher {

    override fun <T : Any> post(event: T) {
        mediator.post(event)
    }

}
